### PR TITLE
ft: create update store item component

### PR DESCRIPTION
**Pull request by tolls-3**
## Description
Create a component that helps sellers to edit products int the store
## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
## Checklist
Remove any items which are not applicable.
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
## Link to Trello card
https://trello.com/c/W8e8DVf6/32-create-edit-item-page

### CODE REVIEW

**Comment by tolls-3** [comment]
[Create edit item page](https://trello.com/c/W8e8DVf6/32-create-edit-item-page)

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "react-router-dom": "^5.1.2",
     "react-scripts": "3.3.0",
     "redux": "^4.0.4",
+    "redux-persist": "^6.0.0",
```
Comment (shaunorpen) [inline package.json]: We're adding this, but not using it as far as I can tell?

File: src/App.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ import CreateStoreForm from './components/createStore/firstView'
 import AddLogoForm from './components/createStore/addLogo'
 import EditProfile from './components/EditProfile/EditProfile'
 import CreateItem from './components/CreateItem'
+import UpdateItem from './components/updateItem'
 // import Inventory from './components/inventory/inventory'
```
Comment (shaunorpen) [inline src/App.js]: Can we delete commented code before merging?

File: src/App.js [hunk 3]
```diff
@@ -42,6 +43,9 @@ function App() {
       <Route path='/createitem'>
         <CreateItem />
       </Route>
+      <Route path='/updateitem/:id' component={UpdateItem}/>
+
```
Comment (shaunorpen) [inline src/App.js]: Is this extra white space necessary?

File: src/components/inventory/inventory.js [hunk 4]
```diff
@@ -68,7 +69,10 @@ const Items = ({ inventory }) => {
       size='small'
       itemLayout='horizontal'
       dataSource={inventory}
-      renderItem={item => (
+      renderItem={item => {
+
```
Comment (shaunorpen) [inline src/components/inventory/inventory.js]: Is this extra white space necessary?

File: src/App.js [hunk 5]
```diff
@@ -42,6 +43,9 @@ function App() {
       <Route path='/createitem'>
         <CreateItem />
       </Route>
+      <Route path='/updateitem/:id' component={UpdateItem}/>
```
Comment (shaunorpen) [inline src/App.js]: Can we create this route using the same syntax as the other routes?

File: src/components/updateItem.js [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+      .get(
+        `https://shopping-cart-eu3-staging.herokuapp.com/api/store/products/${itemId}`
+      )
+      .then(res => {
+        setItem(res.data);
+      });
+  }, [itemId]);
+
+
+const test = item && item.images && item.images.length && item.images.map(i=>{
```
Comment (shaunorpen) [inline src/components/updateItem.js]: This code looks like it's just for testing? Can it be removed before merging?

File: src/components/updateItem.js [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+  return i
+})
+const myArray = test && test.length && test.map((str, index) => ({ url: str, id: index + 1 , name:'image.png'}));
+console.log(myArray)
```
Comment (shaunorpen) [inline src/components/updateItem.js]: Can we remove console.logs before merging?

File: src/components/updateItem.js [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    },
+    {
+      uid: '-3',
+      name: 'image.png',
+     
+      url: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
+    },]
+
+  );
+  console.log(fileList)
```
Comment (shaunorpen) [inline src/components/updateItem.js]: Can we remove this before merging?

File: src/components/updateItem.js [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+
+
+const test = item && item.images && item.images.length && item.images.map(i=>{
+  return i
+})
+const myArray = test && test.length && test.map((str, index) => ({ url: str, id: index + 1 , name:'image.png'}));
+console.log(myArray)
+
+
+  const [fileList, setFileList] = useState(
```
Comment (shaunorpen) [inline src/components/updateItem.js]: This looks like a hard coded file list, perhaps for testing? Can it be removed before merging?

**Review by shaunorpen — APPROVED** [review]
Approving with questions and suggestions, but nothing that impacts functionality.

**Comment by shaunorpen** [comment]
Oops. Didn't notice this was a draft. Sorry!

File: package.json (hunk 1 above)
Comment (tolls-3) [reply package.json]: I have removed it.

File: src/App.js (hunk 3 above)
Comment (tolls-3) [reply src/App.js]: White space has been removed.

File: src/App.js (hunk 5 above)
Comment (tolls-3) [reply src/App.js]: We have to use it in this format in order to pass the id.

File: src/components/updateItem.js (hunk 6 above)
Comment (tolls-3) [reply src/components/updateItem.js]: Removed

File: src/components/updateItem.js (hunk 7 above)
Comment (tolls-3) [reply src/components/updateItem.js]: Done

File: src/components/updateItem.js (hunk 8 above)
Comment (tolls-3) [reply src/components/updateItem.js]: Done

File: src/components/updateItem.js (hunk 9 above)
Comment (tolls-3) [reply src/components/updateItem.js]: Done.

**Comment by DimejiAre** [comment]
@tolls-3 Have you pulled from develop?

**Review by shaunorpen — APPROVED** [review]

**Pull request merged**